src/Config/ObjectModel/Entity.cs

```diff
@@ -62,4 +73,36 @@ public record Entity
         Cache is not null &&
         Cache.Enabled is not null &&
         Cache.Enabled is true;
+
+    /// <summary>
+    /// Helper method to generate the linking entity name using the source and target entity names.
+    /// </summary>
+    /// <param name="source">Source entity name.</param>
+    /// <param name="target">Target entity name.</param>
+    /// <returns>Name of the linking entity.</returns>
+    public static string GenerateLinkingEntityName(string source, string target)
+    {
+        return LINKING_ENTITY_PREFIX + ENTITY_NAME_DELIMITER + source + ENTITY_NAME_DELIMITER + target;
+    }
+
+    /// <summary>
+    /// Helper method to decode the names of source and target entities from the name of a linking entity.
+    /// </summary>
+    public static Tuple<string, string> GetSourceAndTargetEntityNameFromLinkingEntityName(string linkingEntityName)
+    {
+        if (!linkingEntityName.StartsWith(LINKING_ENTITY_PREFIX+ENTITY_NAME_DELIMITER))
+        {
+            throw new Exception("The provided entity name is an invalid linking entity name.");
+        }
+
+        string entityNameWithLinkingEntityPrefix = linkingEntityName.Substring(LINKING_ENTITY_PREFIX.Length + ENTITY_NAME_DELIMITER.Length);
```

Comment:
Why not directly use split function with `$` and collect the 1st and 2nd items from the resulting array?

Reply:
makes sense